docs/self-hosting/self-hosting.md

```diff
@@ -0,0 +1,70 @@
+---
+sidebar_position: 1
+title: Self-Hosting DBOS Apps
+---
+
+Because DBOS is implemented in lightweight TypeScript and Python libraries, you can run your DBOS application anywhere as long as it has a Postgres server to connect to.
+This guide provides information on operating a self-hosted DBOS application.
```

Comment:
I'd expand on the concepts first. It is important people understand what are the DBOS components which require special management (spoiler: versioned workflows & their recovery)

- `you can run your DBOS application anywhere`: a DBOS app is a normal Python / node process that can be managed by any cluster manager (e.g., your local machine, kubernetes)
- What is different about a DBOS app? It has (versioned) workflows and recovery processes.

-> explain workflow versioning
-> explain recovery process
-> Then go over the difference between 1 host and multiple hosts